### PR TITLE
Add NOTICES.txt

### PR DESCRIPTION
**Pull request by doodlesbykumbi**

### CODE REVIEW

File: NOTICES.txt [hunk 1]
```diff
@@ -0,0 +1,395 @@
...
+Section 1: Apache-2.0
+>>> https://github.com/kubernetes/api/tree/acf347b865f2
+>>> https://github.com/kubernetes/apimachinery/tree/19e3f5aa3adc
+>>> https://github.com/kubernetes/kubernetes/tree/v1.9.4
+
+Section 2: MIT
+>>> https://github.com/stretchr/testify/tree/v1.5.1
+
+Section 3: BSD-3-Clause
+>>> https://github.com/evanphx/json-patch
```
Comment (izgeri) [inline NOTICES.txt:18]: this one doesn't have a version

Comment (doodlesbykumbi) [reply NOTICES.txt]: Updated

**Comment by doodlesbykumbi** [comment]
@izgeri 
> The go.mod has 32 included deps, but there's only 5 reflected here - what's the difference?
The difference is that the other dependencies are indirect. I wonder if we should be including indirect licenses too

**Comment by izgeri** [comment]
can you please confirm that none of the `indirect`-marked deps in go.mod are libraries we manually included, and/or that all `indirect`-marked deps are tier 2+ dependencies?

based on [the go docs](https://github.com/golang/go/wiki/Modules#why-does-go-mod-tidy-record-indirect-and-test-dependencies-in-my-gomod) it looks like modules labeled `indirect` are either there because a direct dep is missing a go.mod file itself, so its dependencies are recorded - or because you manually specified a different version of a 2nd tier dep than what a dep requires (if I'm reading it correctly)

I just want to confirm if we're not going to include the indirect deps (which I think is fine) that they are all indeed coming from deps-of-deps, and aren't direct includes / versions that we manually specified / required

**Comment by doodlesbykumbi** [comment]
@izgeri all the `indirect`-marked deps are 2nd tier and none where manually specified by us.

**Pull request merged**